sql/src/main/java/io/crate/execution/engine/collect/sources/InformationSchemaIterables.java

```diff
@@ -115,7 +116,18 @@ public InformationSchemaIterables(final Schemas schemas,
             .flatMap(r -> sequentialStream(new NotNullConstraintIterable(r)))
             .iterator();
 
-        constraints = () -> concat(sequentialStream(primaryKeyConstraints), sequentialStream(notnullConstraints))
+        Iterable<ConstraintInfo> checkConstraints = () ->
+            sequentialStream(relations)
+                .filter(r -> !r.checkConstraints().isEmpty())
```

Comment:
I think the filter is not necessary, `flatMap` won't generate new entries if `checkConstraints` is empty.